ldms/src/ldmsd/ldmsd_config.c

```diff
@@ -209,10 +209,7 @@ ldmsd_sampler_add(const char *cfg_name,
 	sampler->api = api;
 	sampler->libpath = strdup(libpath);
 	sampler->thread_id = -1; /* stopped */
-	if (sampler->api->base.constructor != NULL
-		&& sampler->api->base.destructor != NULL)
-	{
-		/* This plugin is multi-instance capable */
+	if (sampler->api->base.constructor != NULL) {
```

Comment:
I think with the cfgobj_generic and/or plugin_generic you are proposing, these code paths can be combined, specifically config can be combined for sampler and store.